### DISCUSSION TITLE
Text Adjustment

### DISCUSSION THREAD

**Discussion by Guspin** [ImpulseAdventure/GUIslice#449] [Q&A]
I was wondering if there is a way of automatic adjust a string to a text element. I don't know if this feature already exists.
If I didn't explain correctly I want to have a square of a text box and set the string of the text element to adjust to the space that it had.

Sorry for my english, it isn't my first languaje.

**Comment by Pconti31** [comment]
@Guspin Well not sure what you want here.  If you are using the Builder Text strings are automatically sized., making the frame about one pixel more then the size of your text.  This allows you to place them exactly where you want.   

You can of course make the frame larger or smaller using the height and width properties.  In this case you can choose the alignment within the frame by select from the drop down control within the `Text Alignment` property, selecting from:

 - GSLC_ALIGN_MID_LEFT
 - GSLC_ALIGN_MID_MID
 - GSLC_ALIGN_MID_RIGHT
 - GSLC_ALIGN_TOP_LEFT
 - GSLC_ALIGN_TOP_MID
 - GSLC_ALIGN_TOP_RIGHT
 - GSLC_ALIGN_BOT_LEFT
 - GSLC_ALIGN_BOT_MID
 - GSLC_ALIGN_BOT_RIGHT

You can also set how far to indent text within a frame by setting the `Text Margin` property from 0 to whatever you need.
Paul--
